sqsworkers/base.py

```diff
@@ -0,0 +1,276 @@
+            name: the name of the Crew, for logging
+            statsd: statsd client
+            sentry: sentry client
+            max_number_of_messages: passed to self.queue.receive_messages(MaxNumberOfMessages=...)
+            wait_time: passed to self.queue.receive_messages(WaitTimeSeconds=...)
+            polling_interval: How long to wait in between polls on sqs
+            log_level: the logging level for this instance's logger
+        """
+        xor_msg_proc = bool(MessageProcessor) ^ bool(message_processor)
+
+        xor_error_msg = f"message_processor and MessageProcessor arguments are mutually exclusive"
+
+        assert xor_msg_proc, xor_error_msg
+
+        message_processor = message_processor or MessageProcessor
+
+        assert issubclass(
+            message_processor, interfaces.CrewInterface
+        ), f"{message_processor.__name__} does not conform to {interfaces.CrewInterface.__name__}"
+
+        self.message_processor = message_processor
+
+        if MessageProcessor is not None:
+            warnings.warn(
+                os.linesep.join(
+                    [
+                        "MessageProcessor argument will be deprecated in future version",
+                        "please use message_processor instead",
+                    ]
+                ),
+                DeprecationWarning,
+            )
+
+        deprecated = ["workers", "supervisor", "exception_handler_function"]
+
+        for d in deprecated:
+            if locals().get(d) is not None:
+                warnings.warn(
+                    f"The {d} argument is deprecated and will be ignored.",
+                    DeprecationWarning,
+                )
+
+        self.sqs_session = sqs_session
+
+        assert not (
+            sqs_resource is None and queue_name is None
+        ), "You must pass either the queue name or sqs_resource"
+
+        self.queue = self.sqs_resource = (
+            sqs_session.resource(
+                "sqs", region_name=sqs_session.region_name
+            ).get_queue_by_name(QueueName=queue_name)
+            if sqs_resource is None
+            else sqs_resource
+        )
+
+        self.name = name or "crew-{}-{}-{}".format(
+            os.getpid(), (queue_name or next(self._count)), time.time()
+        )
+
+        self.logger = logging.LoggerAdapter(
+            (logging.getLogger() if logger is None else logger),
+            extra={"extra": {"crew.name": self.name}},
+        )
+
+        self.logger.setLevel(log_level)
+
+        statsd = StatsDBase(logger=logger) if statsd is None else statsd
+
+        assert issubclass(
+            statsd.__class__, interfaces.StatsDInterface
+        ), f"{statsd.__class__} does not conform to {interfaces.StatsDInterface.__name__}"
+
+        self.statsd = statsd
+        self.sentry = sentry
+        self.max_number_of_messages = max_number_of_messages
+        self.wait_time = wait_time
+        self.polling_interval = polling_interval
```

Comment:
What if no polling_interval is defined, would we just fail or default to something?

Reply:
it defaults to one https://github.com/goodmanship/sqsworkers/pull/13/files/e828f00747a0b50c9bae1b3649b31fb8c068872f#diff-5dc2307bb18093b73257d7a60ee66acbR89